dlib/rand/rand_kernel_1.h

```diff
@@ -112,25 +112,25 @@ namespace dlib
                 next_gaussian = 0;
             }
 
-            unsigned char get_random_8bit_number (
+            virtual unsigned char get_random_8bit_number (
```

Comment:
This class isn't meant to be inherited from.  The destructor isn't virtual for instance and you will generate object slicing problems by doing this as well.

There is also contract violation now since functions that take a `dlib::rand` expect that object.  For instance, `.clear()` and `.set_seed()` would have the wrong behavior.

The right way to do this would be to just have a separate class.

Reply:
Thanks for the review. I see, I'm going to separate this feature into a new class. I was trying to find a way to reuse the helper functions of the dlib::rand class like get_random_double, any recommendations?